documentation/5_TechStack.md

```diff
@@ -0,0 +1,102 @@
+# Tech Stack
+
+  Please refer package.json for all libraries details
+  TypeScript, Node.js, GraphQL, Express etc..
+
+### Libraries
+
+  Used below Npm library to get longitude and latitude:
+  https://github.com/nchaulet/node-geocoder#readme
+
+  Created API Key in Google Cloud in order to use the above library
+
+### Deployment tools
+  
+  My observations about infrastructure-as-code tooling
+
+  1. Terraform, for me writing Terraform code is new, and my observation is need to write extra code for permissions like iam role policy
+  and I noticed with cdk you can write less code, so I chose cdk tool and more time given happy to try terraform as well, this is the document I am referring to https://learn.hashicorp.com/tutorials/terraform/lambda-api-gateway
+
+  2. CloudFormation / SAM - haven't tried.
+
+  3. Serverless Framework, I have noticed apollo has documentation around serverless framework, for some reason I couldn't find integration with api gateway, I just saw lambda, this is the doc I referred https://www.apollographql.com/docs/apollo-server/deployment/lambda/
+
+  4. AWS CDK is what i have used for deploying all resources to AWS.
+
+### Logger
+
+  Used loggingPlugin. Provided Screenshot from Lambda logs.
+
+  Other loggers:
+  1. https://www.envelop.dev/plugins/use-logger, couldn't set the custom plugin using envelop plugin
+
+  ```js
+  import { envelop, useLogger } from '@envelop/core';
+  plugins: [useLogger({ logFn: () => logger })],
+  ```
+
+  2. graphql-pino-middleware
+
+  ```js
+  import graphqlPinoMiddleware from 'graphql-pino-middleware';
+  create the graphql-pino-middleware
+  const loggerMiddleware = graphqlPinoMiddleware({ logger: log });
+  ```
+
+  3. loggingMiddleware
+  
+  ```js
+  const loggingMiddleware = (req: any, _res: any, next: any) => {
+    console.log('ip:', req.ip);
+    next();
+  };
+  ```
+  If more time given, can explore above loggers as well.
+
+  4. Couldn't use Pino logger, giving pino error like `FinalizationRegistry is not defined`, i have tried registering to Pino middleware as well, didn't work. Also with the latest Pino types, there are tsc errors.
+
+### Traceability
+
+  Tried Open Telemetry, but getting TS errors - couldn't succeed. Left commented code, given more time can explore some more.
+  https://opentelemetry.io/docs/instrumentation/js/getting-started/nodejs/
+  https://github.com/open-telemetry/opentelemetry-js
+
+  ```js
+  const OpentracingPlugin = require('apollo-opentracing').default;
+  ```
+
+### GraphQL Hive
+
+  For monitoring, https://docs.graphql-hive.com
```

Comment:
What would be the pros and cons of choosing GraphQL Hive vs Cloudwatch alarms in the context of this project?

Reply:
With Hive - you can measure or collect data against all your GraphQL operations and generate analytics on the same, not sure if this is possible/out of box feature with Cloudwatch alarms unless you configure it manually. To be honest with you, i didn't use Hive enough to observe more about its features, At my work place GraphQL is just one year old and POC project.

Reply:
That's fair @himabindugit. Appreciate the honesty

Reply:
I have pushed a change integrating with Hive, i am seeing lambda logs related to Hive, exploring more
<img width="832" alt="Screen Shot 2022-08-16 at 12 37 04 PM" src="https://user-images.githubusercontent.com/22843777/184932536-d091d367-3529-4ec8-a21b-5ec16343d1da.png">
e

Reply:
Yay, able to register schema and see operations in Hive, nice UI
<img width="321" alt="Screen Shot 2022-08-16 at 2 07 54 PM" src="https://user-images.githubusercontent.com/22843777/184948954-e0b4b13b-8522-4dc8-bd82-9fa61cf4d7bb.png">
<img width="1272" alt="Screen Shot 2022-08-16 at 2 08 08 PM" src="https://user-images.githubusercontent.com/22843777/184948995-3a09d82e-ce08-4067-8052-58d5ef1aa2a8.png">